src/modules/hip/kernel/log.hpp

```diff
@@ -0,0 +1,188 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+// -------------------- Set 1 - helper kernels --------------------
+template <typename T>
+__device__ void log_hip_compute(T *srcPtr, d_float8 *src_f8, d_float8 *dst_f8)
+{
+    if constexpr (std::is_same<T, schar>::value)
+        rpp_hip_math_add8_const(src_f8, src_f8, (float4)128);
+
+    rpp_hip_log(src_f8, dst_f8);
+}
+
+// -------------------- Set 2 - log kernels --------------------
+template <typename T, typename U>
+__global__ void log_2d_hip_tensor(T *srcPtr,
+                                  uint2 srcStridesNH,
+                                  U *dstPtr,
+                                  uint2 dstStridesNH,
+                                  uint *roiTensor)
+{
+    uint id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8; // width
+    uint id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;       // height
+    uint id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;       // batchsize
+
+    uint *roi = &roiTensor[id_z * 4];
+    uint yBegin = roi[0];
+    uint xBegin = roi[1];
+    uint height = roi[2];
+    uint width = roi[3];
```

Comment:
You could possibly use the struct for this

Reply:
We can use struct "RpptRoiXyzwhd" however order of filling values in testsuite differs, this struct expects x, y, z, roiWidth, roiHeight, roiDepth.
Values filled in ROI is in order: zBegin, yBegin, xBegin, lengthZ, lengthY, lengthX as it will be easy to understand this way for ND tensors.

Thus didn't make this change.

Reply:
@snehaa8 @sampath1117 Not sure how we decided the order of ROI filling. Since it was always xyWH or xyzWHD, I'm assuming that even for the ND ROIs say with begin and length, its always the inner dimension that comes first. (So beginX, beginY, beginZ, lengthX, lengthY, lengthZ)

@snehaa8 If that change is made, you can do all this inside the kernel in one line.
Please receive it in the kernel as uint4 *roiTensor_u4 instead. Then just initialize a local u4 and use it.

```
uint4 roi_u4 = roiTensor_u4[id_z];    // Contains roi in the order of begin, beginX, beginY, lengthX, lengthY
```

Then just use roi_u4.x, roi_u4.y, roi_u4.z, roi_u4.w inside